src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedViewAccessor.cs

```diff
@@ -34,7 +34,7 @@ protected override void Dispose(bool disposing)
             {
                 // Explicitly flush the changes.  The OS will do this for us anyway, but not until after the
                 // MemoryMappedFile object itself is closed.
-                if (disposing && !_view.IsClosed)
+                if (disposing && !_view.IsClosed && CanWrite)
```

Comment:
Does MemoryMappedViewStream.Dispose need the same treatment?

Reply:
good point, I've applied the same change to `MemoryMappedViewStream` as well. PTAL